### DISCUSSION TITLE
Typescript | Prisma. Models and required fields.

### DISCUSSION THREAD

**Discussion by tobiholic** [prisma/prisma#25258] [Q&A]
### Question

I have the following struggle / question:
For a personal project I created the following prisma models:

```
model Job {
  id              Int        @id @default(autoincrement())
  verified        Boolean    @default(false)
  offline         Boolean    @default(false)
  title           String     @db.VarChar(255) //default: varchar(191)
  description     String     @db.Text
  skills          String     @db.Text
  offer           String     @db.Text
  price           Int
  languageEnglish Boolean?   @default(false)
  payment         Payment
  category        Category
  timezone        Timezone   @default(WELTWEIT)
  timeEffort      TimeEffort @default(PROJEKTARBEIT)
  company         Company?   @relation(fields: [companyId], references: [id], name: "Client")
  companyId       Int?       @unique
  likedByUsers    JobLikes[]
  createdAt       DateTime   @default(now())
  updatedAt       DateTime   @updatedAt
}
```

```
model Company {
  id           Int      @id @default(autoincrement())
  name         String   @db.VarChar(255)
  description  String   @db.Text()
  contactFN    String   @db.VarChar(255)
  contactLN    String   @db.VarChar(255)
  contactEmail String
  job          Job?     @relation("Client")
  createdAt    DateTime @default(now())
  updatedAt    DateTime @updatedAt
}
```

```
export interface JobWithCompany extends Job {
  company: Company | null;
}
```

The idea is that a user can create a job via a form. In the form I only ask for the needed information, like description, title and so on.. 
After submit, I want to save the data in a Zustand store to access it on a preview page. Now, when I claim the Zustand store to be a JobWithCompany type, it also requires properties like: id, createdAt, verified.. (all properties with a default value or optional).
What is the best way to solve this? The zustand store should only hold the input values of the user from the form. **Creating a new interface for Zustand feels wrong, but also putting properties like id on optional in the prisma schema feels wrong.** 

Thanks already for the help!! 

Cheers, Tobi

**Comment by ludralph** [comment]
Hi @tobiholic 👋 

Since you are extending `Job` in the `JobWithCompany` interface, it expects all properties from the `Job` model. If you don't want to create a new `JobFormInput` interface, you can use TypeScript's utility types like `Pick` or `Partial` to dynamically extract only the relevant fields from your `Job` model. This way, your form interface is directly tied to the Job model, and it automatically updates if the model changes.
```ts
import { Job } from '@prisma/client'; 

export type JobFormInput = Pick<
  Job,
  'title' | 'description' | 'skills' | 'offer' | 'price' | 'languageEnglish' | 'category' | 'timezone' | 'timeEffort' | 'companyId'
>;
```

With this, your `JobFormInput` stays in sync with the `Job` model without having to manually update it when fields are added or changed.

**Reply by tobiholic** [reply]
Hi @ludralph!
Thanks for your help, really appreciated. Thats what I looked for! Meanwhile, I went on in my project creating an new interface for the FormState, decided to seperate the concern of FormState and Prisma/Database state. But still I am not 100% convinced, so I will check your suggestion. 

Have a good one!
